src/vs/editor/common/config/editorOptions.ts

```diff
@@ -4814,7 +4814,7 @@ export const EditorOptions = {
 	)),
 	formatOnType: register(new EditorBooleanOption(
 		EditorOption.formatOnType, 'formatOnType', false,
-		{ description: nls.localize('formatOnType', "Controls whether the editor should automatically format the line after typing.") }
+		{ description: nls.localize('formatOnType', "Controls whether the editor should automatically format the line after typing. A formatter must be available and the formatter should be able to format a range in a document.") }
```

Comment:
> formatter should be able to format a range in a document.

That's actually not 100% correct. Format on type requires a special formatter type (`OnTypeFormattingEditProvider`) and not just a range formatter.  That's quite technical and I would rephrase that format on type requires one to have a corresponding formatter extension that support this

Reply:
I have updated it. :)